tableauserverclient/models/schedule_item.py

```diff
@@ -0,0 +1,239 @@
+import xml.etree.ElementTree as ET
+from .interval_item import *
+from .property_decorators import property_is_enum, property_not_nullable
+from .. import NAMESPACE
+
+
+class ScheduleItem(object):
+    class Type:
+        Extract = "Extract"
+        Subscription = "Subscription"
+
+    class ExecutionOrder:
+        Parallel = "Parallel"
+        Serial = "Serial"
+
+    class State:
+        Active = "Active"
+        Suspended = "Suspended"
+
+    def __init__(self, name, priority, schedule_type, execution_order, interval_item):
+        self._created_at = None
+        self._end_schedule_at = None
+        self._execution_order = None
+        self._frequency = None
+        self._id = None
+        self._name = None
+        self._next_run_at = None
+        self._priority = None
+        self._schedule_type = None
+        self._state = None
+        self._updated_at = None
+        self.interval_item = interval_item
+
+        # Invoke setter
+        self.execution_order = execution_order
+        self.name = name
+        self.priority = priority
+        self.schedule_type = schedule_type
+
+    @property
+    def created_at(self):
+        return self._created_at
+
+    @property
+    def end_schedule_at(self):
+        return self._end_schedule_at
+
+    @property
+    def execution_order(self):
+        return self._execution_order
+
+    @execution_order.setter
+    @property_is_enum(ExecutionOrder)
+    def execution_order(self, value):
+        self._execution_order = value
+
+    @property
+    def frequency(self):
+        return self._frequency
+
+    @property
+    def id(self):
+        return self._id
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    @property_not_nullable
+    def name(self, value):
+        self._name = value
+
+    @property
+    def next_run_at(self):
+        return self._next_run_at
+
+    @property
+    def priority(self):
+        return self._priority
+
+    @priority.setter
+    def priority(self, value):
+        if value < 1 or value > 100:
```

Comment:
Ok, done.

```
def property_is_int(min, max=2**31):
    def property_type_decorator(func):
        @wraps(func)
        def wrapper(self, value):
            if min is None:
                return func(self, value)

            if value < min or value > max:
                error = "Invalid priority defined: {}.".format(value)
                raise ValueError(error)

            return func(self, value)

        return wrapper

    return property_type_decorator
```

This means that passing just None will skip validation, otherwise it's min/max.

Technically to make this signature work, Max has a default value of 2**31